collect_app/src/main/java/org/odk/collect/android/tasks/InstanceSyncTask.java

```diff
@@ -107,7 +107,9 @@ protected String doInBackground(Void... params) {
                     while (instanceCursor.moveToNext()) {
                         String instanceFilename = instanceCursor.getString(
                                 instanceCursor.getColumnIndex(InstanceColumns.INSTANCE_FILE_PATH));
-                        if (candidateInstances.contains(instanceFilename)) {
+                        String instanceStatus = instanceCursor.getString(
```

Comment:
@nribeka we can't do that please notice that candidateInstance contains all instances (those submitted as well). So we need to have a cursor that contains all to remove submitted instances https://github.com/opendatakit/collect/pull/1126/files#diff-c6b798e69cb47b6c50f0e791fb67ab1dR113